src/mono/mono/metadata/assembly-load-context.c

```diff
@@ -32,38 +47,41 @@ mono_alc_init (MonoAssemblyLoadContext *alc, MonoDomain *domain, gboolean collec
 }
 
 static MonoAssemblyLoadContext *
-mono_alc_create (MonoDomain *domain, gboolean is_default, gboolean collectible)
+mono_alc_create (gboolean collectible)
 {
 	MonoAssemblyLoadContext *alc = NULL;
 
-	mono_domain_alcs_lock (domain);
-	if (is_default && domain->default_alc)
-		goto leave;
+	alcs_lock ();
```

Comment:
I think we can take this slightly later, just around the part where we modify `alcs`? I don't think `mono_alc_init` does anything that needs it.